### DISCUSSION TITLE
Log rotation not working correctly on Windows

### DISCUSSION THREAD

**Discussion by nick-randal** [rabbitmq/rabbitmq-server#2479] [Other] [answered]
settings from rabbitmq.conf

``` ini
log.file = rmq.log
log.file.rotation.size = 104857600
log.file.rotation.size = 4
```

All the files are continually updated and seem to be swapping log text back and forth between them on a constant basis.

![image](https://user-images.githubusercontent.com/6055507/96787542-ed8c2c80-13bf-11eb-96d5-70624f1096e1.png)

The only solution right now is to disable rotation. I also tried rotation by D0 but that did the exact same thing.

![image](https://user-images.githubusercontent.com/6055507/96787794-5d021c00-13c0-11eb-8f80-4852e3fadd93.png)

supported by Erland 22.2 on windows server 2019

**Comment by michaelklishin** [comment]
@nick-randal your config file has a duplicate key. `rabbitmq-diagnostics environment` will print effective configuration. The list of applications only tells us the versions. You are seven patch releases behind, so it would be great if testing was done using `3.8.9`.

@lukebakken we last had to deal with log rotation on Windows in https://github.com/rabbitmq/rabbitmq-server/pull/2078, and Lager version hasn't changed since then. I could not find any new relevant issues in Lager.

**Reply by nick-randal** [reply]
Thank you for the response, this was an embarrassing copy / paste mistake on my part.

**Comment by michaelklishin** [comment, marked as answer]
Moved to a discussion until we have a set of steps to reproduce with `3.8.9`. I could not find any recent reports of issues with log rotation on Windows for either RabbitMQ or Lager (the logging library we use).

**Comment by lukebakken** [comment]
Log file rotation appears to work correctly on Windows. I'm using Erlang 23.1.1 and RabbitMQ 3.8.9. Note that the configuration file in the original report has two duplicate keys. I'm using debug logging to increase the amount of messages and force rotation.

`%AppData%\RabbitMQ\rabbitmq.conf`:

```
log.file.level = debug
log.file.rotation.count = 5
log.file.rotation.size = 1024
```

Log files after starting RabbitMQ a couple times:

```
 Directory of C:\Users\x-lbakken\AppData\Roaming\RabbitMQ\log

10/22/2020  04:32 PM    <DIR>          .
10/22/2020  04:26 PM    <DIR>          ..
10/22/2020  04:26 PM    <DIR>          log
10/22/2020  04:35 PM             7,417 rabbit@DSCH-WIN.log
10/22/2020  04:31 PM            36,867 rabbit@DSCH-WIN.log.0
10/22/2020  04:31 PM             3,808 rabbit@DSCH-WIN.log.1
10/22/2020  04:31 PM            34,340 rabbit@DSCH-WIN.log.2
10/22/2020  04:30 PM             7,485 rabbit@DSCH-WIN.log.3
10/22/2020  04:28 PM             3,773 rabbit@DSCH-WIN.log.4
10/22/2020  04:30 PM               136 rabbit@DSCH-WIN_upgrade.log
```

Note that a strict 1024-byte cutoff is not possible due to log message buffering.
